build/gen_stub.php

```diff
@@ -5156,6 +5157,11 @@ function generateClassEntryCode(FileInfo $fileInfo, array $allConstInfos): strin
 
 /** @param FuncInfo[] $funcInfos */
 function generateFunctionEntries(?Name $className, array $funcInfos, ?string $cond = null): string {
+    // No need to add anything if there are no function entries for a class
+    if ($className !== null && $funcInfos === []) {
```

Comment:
Nit: Right now the same condition for global functions is here: https://github.com/php/php-src/blob/f78d5cfcd2fe06ddd6da33ff880c6823072adc1b/build/gen_stub.php#L5106, so it would be possible to remove the `$className !== null` condition from here along with the referenced `if` to simply the code a little bit.

But I'm also fine as-is if that's what you prefer.

Reply:
Good catch, done